src/test/java/com/solvd/movie/service/impl/EsMovieServiceTest.java

```diff
@@ -0,0 +1,128 @@
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void retrieveAllByYear() {
+        EsMovie movie = MovieFactory.getEsMovie();
+        SearchCriteria criteria = new SearchCriteria();
+        criteria.setYear(2023);
+        Mockito.when(this.esMovieRepository.findAllByYear(movie.getYear()))
+                .thenReturn(Flux.just(movie));
+        Flux<EsMovie> movies = this.esMovieService.retrieveAllByCriteria(
+                criteria
+        );
+        StepVerifier.create(movies)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void retrieveAllByName() {
+        EsMovie movie = MovieFactory.getEsMovie();
+        SearchCriteria criteria = new SearchCriteria();
+        criteria.setName("Name");
+        Mockito.when(this.esMovieRepository.findAllByName(movie.getName()))
+                .thenReturn(Flux.just(movie));
+        Flux<EsMovie> movies = this.esMovieService.retrieveAllByCriteria(
+                criteria
+        );
+        StepVerifier.create(movies)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void retrieveAllByNameAndeYear() {
+        EsMovie movie = MovieFactory.getEsMovie();
+        SearchCriteria criteria = new SearchCriteria();
+        criteria.setName("Name");
+        criteria.setYear(2023);
+        Mockito.when(this.esMovieRepository.findAllByNameAndYear(
+                        movie.getName(),
+                        movie.getYear()
+                ))
+                .thenReturn(Flux.just(movie));
+        Flux<EsMovie> movies = this.esMovieService.retrieveAllByCriteria(
+                criteria
+        );
+        StepVerifier.create(movies)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void create() {
+        EsMovie movie = MovieFactory.getEsMovie();
+        Mockito.when(this.esMovieRepository.save(movie))
+                .thenReturn(Mono.just(movie));
+        Mono<EsMovie> createdMovie = this.esMovieService.create(movie);
+        StepVerifier.create(createdMovie)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void update(){
+        EsMovie movie = MovieFactory.getEsMovie();
+        Mockito.when(this.esMovieRepository.save(movie))
+                .thenReturn(Mono.just(movie));
+        Mockito.when(this.esMovieRepository.findById(movie.getId()))
+                .thenReturn(Mono.just(movie));
+        Mono<EsMovie> updatedMovie = this.esMovieService.update(movie);
+        StepVerifier.create(updatedMovie)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void delete(){
```

Comment:
@hizmailovich follow test naming conventions

read [this](https://stackoverflow.com/questions/155436/unit-test-naming-best-practices) and [this](https://www.yegor256.com/2023/01/19/layout-of-tests.html)